### ISSUE TITLE
Does BarchNorm need different weight decay?

### ISSUE THREAD

**Issue by ntubiolin** [ntubiolin/xcos#1]
In original Insight Pytorch
```
self.optimizer_fr = optim.SGD([
                                    {'params': paras_wo_bn + [self.head.kernel], 'weight_decay': 5e-4},
                                    {'params': paras_only_bn}
                                ], lr = conf.lr, momentum = conf.momentum)
```
, the optimizer will apply weight_decay on batch norm layer.
However, given the current template, it seems that we cannot directly apply optimizers on part of the model, so I decide to  apply no weight decay on it (although this ignorance may lead to [the problem](https://discuss.pytorch.org/t/weight-decay-in-the-optimizers-is-a-bad-idea-especially-with-batchnorm/16994/3)). 
I may modify the BasePipeline._setup_optimizers() in the future.
